src/entities/entity.cpp

```diff
@@ -0,0 +1,115 @@
+#include "entity.h"
+
+int Entity::_id = 0;
+
+Entity::Entity(EntityType type) {
+    _entityType = type;
+    _id++;               // global id +1
+    _entityPriority = 0; // default priority is 0
+    _loopFrequency = 60; // default loop frequency is 60
+    _isEnabled = true;   // enabling by default
+    _loopEnabled = true; // enabling loop by default
+}
+
+void Entity::run(){
```

Comment:
This is the core:
Call **initialization()** function first (use it to initialize pointers and vars), and while the thread is enabled it calls **run()** (use it to your module core). At the end, it calls **finalization()** (use it to destroy used pointers or to end connections).